### PR TITLE
Improve Test Compiling

### PR DESCRIPTION
**Pull request by AlbertoBrusa**
- [x] Attempt to use @swc/jest as a test runner instead of ts-jest
- [x] Put functionality under --swc flag as to not create a breaking change 
- [x] Test speed differences between various compiler configurations:
esmView test run locally:
previously took: 163s 
Now takes: 89s 
Overall test times seem to be down anywhere between 5-10 minutes (varies run to run)
- [x] Update docs
- [x] Fix test command

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -36,7 +36,7 @@ jobs:
       - name: 'Build internal prerequisites'
         run: yarn workspace @modular-scripts/workspace-resolver build
       - name: Run Tests
-        run: yarn test --coverage
+        run: yarn test --coverage --regex="./packages/*"
```
Comment (AlbertoBrusa) [inline .github/workflows/test.yml:39]: yarn test won't work for us, cause most of our packages aren't actual modular packages

File: package.json [hunk 2]
```diff
@@ -14,7 +14,7 @@
     "lint:fix": "yarn lint --fix",
     "create-modular-react-app": "ts-node packages/create-modular-react-app/src/cli.ts",
     "modular": "ts-node packages/modular-scripts/src/cli.ts",
-    "test": "cross-env NODE_OPTIONS=--max_old_space_size=5120 yarn modular test --regex --watchAll false --runInBand --env node --transformIgnorePatterns \"node_modules/(?!rollup)/\"",
+    "test": "cross-env NODE_OPTIONS=--max_old_space_size=5120 yarn modular test --swc --runInBand --env=node",
```
Comment (AlbertoBrusa) [inline package.json:17]: Over time between test bugs and trying to make things work I must have messed up our default test command - should be alright now

File: packages/modular-scripts/package.json [hunk 3]
```diff
@@ -37,14 +37,15 @@
...
+    "@swc/core": "^1.3.46",
+    "@swc/jest": "^0.2.26",
     "@types/case-sensitive-paths-webpack-plugin": "^2.1.6",
     "@types/lodash": "^4.14.191",
     "@types/micromatch": "4.0.2",
     "@types/node": "^18.11.18",
     "@types/npmcli__arborist": "^5.6.0",
     "@types/postcss-normalize": "^9.0.1",
     "@types/shell-quote": "^1.7.1",
-    "@types/webpack": "^5.28.0",
```
Comment (AlbertoBrusa) [inline packages/modular-scripts/package.json:47]: not sure why this got removed? - don't remember doing it. 
nothing broke so I'm wondering if the types came with webpack already and this was unnecessary

File: packages/modular-scripts/src/__tests__/test.test.ts [hunk 4]
```diff
@@ -167,6 +167,19 @@ describe('Modular test command', () => {
         'Ran all test suites matching /packages\\/a/i with tests matching "should add two numbers unique"',
       );
     });
+    it('runs all tests when no options are provided', () => {
```
Comment (AlbertoBrusa) [inline packages/modular-scripts/src/__tests__/test.test.ts:170]: This wasn't actually happening  and we didn't spot it, so added a test

**Review by cristiano-belloni — APPROVED** [review]

**Pull request merged**